packages/abstractions/src/utils/guidUtils.ts

```diff
@@ -4,17 +4,38 @@
  * See License in the project root for license information.
  * -------------------------------------------------------------------------------------------
  */
-import { Guid } from "guid-typescript";
+export type Guid = string;
+
+const guidValidator = new RegExp("^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$", "i");
+
+/**
+ * Checks if the string is a valid GUID.
+ * @param source the string to check.
+ * @returns unchanged string if it is a valid GUID; otherwise, undefined.
+ */
+export const parseGuidString = (source?: string) => {
+	if (source && guidValidator.test(source)) {
+		return source;
+	}
+	return undefined;
+};
+
+/**
+ * Generates a GUID.
+ * @returns a GUID.
+ */
+export const createGuid = () => [gen(2), gen(1), gen(1), gen(1), gen(3)].join("-");
 
 /**
- * Parses a string into a Guid object.
- * @param source The source string.
- * @returns The Guid object.
+ * Generates a part of a GUID.
+ * @param count the number of 2 byte chunks to generate.
+ * @returns a part of a GUID.
  */
-export function parseGuidString(source?: string): Guid | undefined {
-	if (source && Guid.isGuid(source)) {
-		return Guid.parse(source);
-	} else {
-		return undefined;
+export const gen = (count: number) => {
```

Comment:
can you provide more context on why does this function need to be exported?

Reply:
Hello! I agree, exporting here makes no sense, should I create a PR to delete it?

Reply:
Please go ahead

Reply:
Please go ahead

Reply:
Done, https://github.com/microsoft/kiota-typescript/pull/1481